src/lib/reflect.ts

```diff
@@ -585,6 +586,17 @@ export class ReflectedGenericRef extends ReflectedTypeRef<format.RtGenericType>
     override matchesValue(value: any, errors?: Error[], context?: string): boolean {
         return this.baseType.matchesValue(value, errors, context);
     }
+
+    matchesRef(that: ReflectedGenericRef): boolean {
+        // return deepEqual(this, that) // is this too lazy?
+        return this.baseType?.ref === that.baseType?.ref
+            && this.typeParameters.length === that.typeParameters.length
+            && this.typeParameters.every((p, i) => deepEquals(p, that.typeParameters[i]));
+            // TODO: do we really need to `matchRef(..)` all the way down?
```

Comment:
Yes, take for instance: `Foo<Bar<Baz>>` or `Foo<string | (Bar<Baz> & { brand: symbol })`